add.go

```diff
@@ -518,6 +518,11 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
 				go func() {
 					var cloneDir, subdir string
 					cloneDir, subdir, getErr = define.TempDirForURL(tmpdir.GetTempDir(), "", src)
+					defer pipeWriter.Close()
+					defer wg.Done()
```

Comment:
Swap these two lines - we want the waitgroup's `Wait()` to wait until after the `pipeWriter` is closed, as before.